test/e2e_node/pod_construction.go

```diff
@@ -0,0 +1,325 @@
+/*
```

Comment:
let's either have a folder "container_lifecycle_utils" or prefix this file with container_lifecycle. Functions here are very specific to how we do lifecycle testing